pkg/jobs/update.go

```diff
@@ -171,6 +171,29 @@ WHERE id = $1
 	if err := updateFn(u.txn, md, &ju); err != nil {
 		return err
 	}
+
+	// a job status is considered updated if:
+	//  1. the status of the updated metadata is not empty
+	//  2. the status of the updated metadata is not equal to old status
+	//  3. the status of the updated metadata and the old status is running
+	// #1 should be sufficient to determine whether a status change has happened
+	// as the status field of ju.md is not empty only when JobMetadata.UpdateStatus is
+	// called, and this is only called in places where a status change is happening.
+	// Since this may not be in the case in the future we add condition #2. #3 is
+	// required when a job starts because it may already have a "running" status.
+	//
+	if ju.md.Status != "" &&
+		(ju.md.Status != status || (ju.md.Status == StatusRunning && status == StatusRunning)) {
```

Comment:
sorry, i should have said this earlier: I'm not following why `ju.md.Status == StatusRunning && status == StatusRunning` implies a status update?

Reply:
This is so that we can capture a "job started"  status change. Im not sure the historical reasonings for why the status change looks like this, but a job can only "start" if the existing status of the job is `StatusPending` or `StatusRunning`, so this condition is expected when a job "starts" but is already in the running status.

https://github.com/cockroachdb/cockroach/blob/a6f53a802cafa063e906c18d85f57c13b2a98315/pkg/jobs/jobs.go#L253